### DISCUSSION TITLE
Provide a Discord Server for Support and Community-Building

### DISCUSSION THREAD

**Discussion by Frier03** [zauberzeug/nicegui#550] [Ideas / Feature Requests]

**Comment by falkoschindler** [comment]
[Referring to the original title] Please see #272 for the ongoing development of a search-as-you-type function for `ui.select` (assuming this is what you are referring to).

Let's focus on the Discord server here.

**Comment by rodja** [comment]
Over at YouTube we also have users asking about a Discord server: https://www.youtube.com/watch?v=HiNNe4Q32U4&lc=UgyRcZCOZ9i5z6GuDcJ4AaABAg

**Reply by falkoschindler** [reply]
And on Reddit: https://www.reddit.com/r/nicegui/comments/121ux79/discord_pls/

**Comment by AdamC03** [comment]
Personally I just see it as the next step in furthering the libs experience. I can tell the developers have put a lot of work into creating something so seem less and easy to use, taking one look at commit history will show you their dedication and I am all here for it. When a discord server is created I believe the community it would foster can be a large boost for niceGUI usage/implementation. It will be far more attractive to newcomers with the knowledge that a large discord community is willing to help them learn.

**Comment by rodja** [comment]
In general launching our own Discord server for community building is a great idea. While I expect the moderation workload will be quite high, the possibilities of real-time communication should not be underestimated. Besides more collaboration opportunities and enhanced community building it would definitely further increase engagement. Also the voice/video chat capabilities may be quite helpful to quickly help out each other. 

However, there are also downsides which we need to consider. The discussions on discord tend to get quite fragmented. Thats what we like about GitHub Issues/Discussion and Stackoverflow. They provide a good framework to keep everything orderly. Also we would lose visibility: Discord discussions are not as easy to be searched and indexed.
Therefore it's maybe better to stick with GitHub discussions in the current state. It ensures that valuable insights and conversations are not lost and accessable to a broader audience. What do you think?

**Comment by falkoschindler** [comment]
As a disclaimer: I don't have personal experience with Discord, so my judgement might be a little wrong. But it might be just what we need to handle all the communication about NiceGUI. Let me explain:

I see the current situation as follows: @rodja and I, the authors and main developers of NiceGUI, try to read and answer every question at GitHub, Reddit, StackOverflow, and YouTube, while fixing bugs, developing new features, and working on other projects at Zauberzeug. So every attempt to engage more users to help each other could reduce the workload on our side and free up resources to be used for new features. From this perspective, I'd support opening a Discord server.

In an ideal scenario users would use StackOverflow and Discord for asking questions which experienced NiceGUI users can answer. When feedback from the developers is needed, a GitHub discussion, issue or pull request is the next step, since they are not necessarily following the discussion on Discord.

**Reply by AdamC03** [reply]
This is the ideal I see aswell

**Comment by rodja** [comment]
Ok. We will create a Discord server. Hopefully there will be enough people to help each other out. @falkoschindler and I have not the time to be around all the time. But let's try and see. Experimentation before "over thinking" :-)

**Comment by falkoschindler** [comment]
We just set up the NiceGUI Discord server: https://discord.gg/TEpFeAaF4f
It's still more or less a draft and needs some more setup before announcing it to the public.

Who would like to help us with that? We would provide admin rights and let you configure the server until it's nice and cozy.

**Reply by AdamC03** [reply]
joined up

**Reply by Frier03** [reply]
joined! :D

**Comment by falkoschindler** [comment]
The Discord server is up and running. Therefore I'll close this discussion as completed. ✔️
